test/dsets.c

```diff
@@ -4041,7 +4041,7 @@ test_nbit_compound_2(hid_t file)
             power               = HDpow(2.0F, (double)(precision[1] - 1));
             orig_data[i][j].a.c = (char)(((long long)HDrandom() % (long long)power) << offset[1]);
             power               = HDpow(2.0F, (double)(precision[2] - 1));
-            orig_data[i][j].a.s = (short)(-((long long)HDrandom() % (long long)power) << offset[2]);
+            orig_data[i][j].a.s = (short)(((long long)HDrandom() % (long long)power) << offset[2]);
```

Comment:
Hard to tell from the current code what would be best.  :-/  I would suggest keeping the negative sign and putting an extra pair of parentheses around the value that's been left-shifted.   That would keep a negative value in the test and should clear the warning up.

Reply:
I'm not sure I understand your suggestion. Extra parens where?

Unlike many of my other patches, this is not for a compilation warning, it's a runtime bug.  In C, `x << y` is (obnoxiously) undefined behaviour if `x < 0`. Decent explanation: https://stackoverflow.com/questions/36751538/ubsan-reports-875-7-as-undefined-behaviour

I suppose we could cast the negative value to unsigned, that would probably preserve the behaviour the author expected.

Reply:
Basically: instead of (short)(- x << y), how about (short)(- (x << y))

Reply:
Oh I see!  Will verify...